hyperopt/spark.py

```diff
@@ -64,8 +66,13 @@ def __init__(self, parallelism=None, timeout=None, spark_session=None):
                               information, visit the documentation for PySpark.
         """
         super(SparkTrials, self).__init__(exp_key=None, refresh=False)
-        if timeout is not None and (not isinstance(timeout, numbers.Number) or timeout <= 0):
-            raise Exception("timeout argument should be None or a positive value.")
+        if not _have_spark:
```

Comment:
In which scenario will ` _have_spark == False` here (if we don't consider anyone manually setting it)? If `pyspark` is not available, `ModuleNotFoundError` will occur, and the class `SparkTrials` cannot be called since you cannot import it.

Reply:
I ran into this when I tried to run the tests with the new --no-spark flag.  When I did that, the import of SparkTrials in the hyperopt/__init__.py file failed.  This change allows the file to be imported when Spark is not available, but causes a failure when a user tries to use SparkTrials.

Reply:
Oh, I see. We want to enable users to `import hyperopt` (which include `from .spark import SparkTrials`) without `pyspark`. So we use the `try: ... except` block. And we also want to explicitly throw an exception when they call `SparkTrials` without `pyspark`. I didn't notice this issue because I didn't realize this is used as a library. Thanks for fixing this!